### PR TITLE
pipeline jsonschema

### PR DESCRIPTION
**Pull request by geoberle**
### What this PR does

added a schema definition for pipeline.yaml files
no validation has been added yet. the mgmt-pipeline.yaml has been adapted to this schema.

Jira: 
Link to demo recording:

### CODE REVIEW

**Review by janboll — APPROVED** [review]

File: tooling/templatize/internal/end2end/e2e_test.go [hunk 1]
```diff
@@ -134,14 +135,15 @@ func TestE2EShell(t *testing.T) {
 		t.Skip("Skipping end-to-end tests")
 	}
 
-	tmpDir := t.TempDir()
+	tmpDir, err := filepath.EvalSymlinks(t.TempDir())
```
Comment (geoberle) [inline tooling/templatize/internal/end2end/e2e_test.go:138]: this is an unrelated fix: tmpdir on mac involved some symlinked oddities :(
